File: engine/model/src/main/scala/tools/packet/cashshop/operation/ShowCashInventory.scala
package tools.packet.cashshop.operation

import client.inventory.Item
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowCashInventory(private var _accountId: Int, private var _items: java.util.List[Item], private var _storageSlots: Byte, private var _characterSlots: Short) extends PacketInput {
  def accountId: Int = _accountId

  def items: java.util.List[Item] = _items

  def storageSlots: Byte = _storageSlots

  def characterSlots: Short = _characterSlots

  override def opcode(): SendOpcode = SendOpcode.CASHSHOP_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/cashshop/operation/ShowCashInventory.java
package tools.packet.cashshop.operation;

import java.util.List;

import client.inventory.Item;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowCashInventory(Integer accountId, List<Item> items, Byte storageSlots,
                                Short characterSlots) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.CASH_SHOP_OPERATION;
   }
}